sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/TLSEvaluationStage.java

```diff
@@ -0,0 +1,291 @@
+                .build();
+    }
+
+    private void storeSecrets(
+            PodInfoBuilder podInfoBuilder, String certPEM, String privateKeyPEM) throws IOException, SecretsException {
+
+        LOGGER.info(String.format("Creating new secret: %s", getSecretPath(podInfoBuilder, "certificate-pem")));
+
+        secretsClient.create(
+                getSecretPath(podInfoBuilder, "certificate-crt"),
+                buildSecret(certPEM, "PEM encoded certificate"));
+
+        LOGGER.info(String.format("Creating new secret: %s", getSecretPath(podInfoBuilder, "private-key")));
+
+        secretsClient.create(
+                getSecretPath(podInfoBuilder, "private-key"),
+                buildSecret(privateKeyPEM, "PEM encoded private key"));
+    }
+
+    private Secret buildSecret(String value, String description) {
+        return new Secret.Builder()
+                .value(value)
+                .author(serviceName)
+                .description(description)
+                .build();
+    }
+
+    private String getSecretPath(PodInfoBuilder podInfoBuilder, String name) {
+        return String.format("%s/%s/%s", serviceName, podInfoBuilder.getPodInstance().getName(), name);
+    }
+
+    private String privateKeyToPem(PrivateKey privateKey) throws IOException {
+
+        StringWriter stringWriter = new StringWriter();
+
+        PemWriter pemWriter = new PemWriter(stringWriter);
+        pemWriter.writeObject(new JcaMiscPEMGenerator(privateKey));
+        pemWriter.flush();
+
+        return stringWriter.toString();
+
+    }
+
+    private String certificateToPem(X509Certificate certificate) throws CertificateEncodingException, IOException {
+
+        StringWriter stringWriter = new StringWriter();
+
+        PemWriter pemWriter = new PemWriter(stringWriter);
+        pemWriter.writeObject(new JcaMiscPEMGenerator(certificate));
+        pemWriter.flush();
+
+        return stringWriter.toString();
+
+    }
+
+    private byte[] generateCSR(KeyPair keyPair) throws IOException, OperatorCreationException {
+
+        X500NameBuilder nameBuilder = new X500NameBuilder();
+        nameBuilder.addRDN(BCStyle.CN, "testing");
+        X500Name name = nameBuilder.build();
+
+        ExtensionsGenerator extensionsGenerator = new ExtensionsGenerator();
+
+        extensionsGenerator.addExtension(
+                Extension.keyUsage, true, new KeyUsage(KeyUsage.digitalSignature));
+
+
+        extensionsGenerator.addExtension(
+                Extension.extendedKeyUsage,
+                true,
+                new ExtendedKeyUsage(
+                        new KeyPurposeId[] {
+                                KeyPurposeId.id_kp_clientAuth,
+                                KeyPurposeId.id_kp_serverAuth }
+                ));
+
+        GeneralNames subAtlNames = new GeneralNames(
+                new GeneralName[]{
+                        new GeneralName(GeneralName.dNSName, "test.com"),
```

Comment:
Is this intentional? Looks like `test.com` and `127.0.0.1` are accidentally copied from the test code further below?